### PR TITLE
Implementing Bigtable.sample_row_keys().

### PR DESCRIPTION
**Pull request by dhermes**

### CODE REVIEW

File: gcloud/bigtable/table.py [hunk 1]
```diff
@@ -145,3 +147,41 @@ def delete(self):
         client = self._cluster._client
         # We expect a `._generated.empty_pb2.Empty`
         client._table_stub.DeleteTable(request_pb, client.timeout_seconds)
+
+    def sample_row_keys(self):
+        """Read a sample of row keys in the table.
+
+        The returned row keys will delimit contiguous sections of the table of
+        approximately equal size, which can be used to break up the data for
+        distributed tasks like mapreduces.
```
Comment (tseaver) [inline gcloud/bigtable/table.py]: The API doesn't take parameters to shape how the sampling is done?

Comment (dhermes) [inline gcloud/bigtable/table.py]: The service [takes][1] a `SampleRowKeysRequest` and the [only field][2] in that message definition is `table_name`.

[1]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/5d84776473347513db0b6c44e4f8503837a0d9fa/bigtable-protos/src/main/proto/google/bigtable/v1/bigtable_service.proto#L44-L46
[2]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/5d84776473347513db0b6c44e4f8503837a0d9fa/bigtable-protos/src/main/proto/google/bigtable/v1/bigtable_service_messages.proto#L89-L93

Comment (tseaver) [inline gcloud/bigtable/table.py]: That seems very strange:  how does the back-end know how many slices the caller wants to create?

Comment (dhermes) [inline gcloud/bigtable/table.py]: The whole idea is that the sample is used to gauge how much stuff is in the table and not much else.

> returned row keys will delimit contiguous sections of the table of approximately equal size ([ref][1], [ref][2])

[1]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/6652d799b564eb8138ebac6a1003371a57fb359b/bigtable-protos/src/main/proto/google/bigtable/v1/bigtable_service.proto#L40-L43
[2]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/6652d799b564eb8138ebac6a1003371a57fb359b/bigtable-protos/src/main/proto/google/bigtable/v1/bigtable_service_messages.proto#L97-L103

Comment (dhermes) [inline gcloud/bigtable/table.py]: Maybe @lesv can correct / augment my understanding?

Comment (lesv) [inline gcloud/bigtable/table.py]: I asked Solomon, Doug or Phanee to comment -- I was told (long ago) that HBase (Map/Reduce) does something with this to partition things up.

Comment (lesv) [inline gcloud/bigtable/table.py]: I heard from Doug -- It doesn't know how many slices, it gives you some starting split points, your code can resplit or not for the workers you have available.

Comment (dhermes) [inline gcloud/bigtable/table.py]: @tseaver SGTY?

Comment (tseaver) [inline gcloud/bigtable/table.py]: I guess so.  I'm seriously puzzled how the back-end expects app developers to use the output from this API.

**Comment by dhermes** [comment]
@tseaver Shall we go ahead and merge?

I think it is just an "esoteric" method needed to emulate HBase. I'm fine with its existence.

**Comment by tseaver** [comment]
Fine to merge.

**Pull request merged**